daml/Marketplace/Custody/Service.daml

```diff
@@ -77,62 +43,22 @@ template Service
-        with
-          openAllocationAccountRequestCid : ContractId Custody.OpenAllocationAccountRequest
-        do
-          Custody.OpenAllocationAccountRequest{nominee; accountId; observers} <- fetchAndArchive openAllocationAccountRequestCid
-
-          let account = Account with provider; owner = customer; id = accountId
-          createOrLookup AllocationAccountRule with ..
-
-      nonconsuming CloseAccount : ()
-        with
-          closeAccountRequestCid : ContractId Custody.CloseAccountRequest
-        do
-          closeAccountRequest <- fetchAndArchive closeAccountRequestCid
-          (assetSettlementRuleCid, _) <- fetchByKey @AssetSettlementRule closeAccountRequest.accountId
-          archive assetSettlementRuleCid
-
-      nonconsuming CloseAllocationAccount : ()
+      nonconsuming Deposit : ContractId AssetDeposit
         with
-          closeAllocationAccountRequestCid : ContractId Custody.CloseAllocationAccountRequest
+          depositRequestCid : ContractId Custody.DepositRequest
         do
-          closeAccountRequest <- fetchAndArchive closeAllocationAccountRequestCid
-          (allocationAccountRuleCid, _) <- fetchByKey @AllocationAccountRule closeAccountRequest.accountId
-          archive allocationAccountRuleCid
+          Custody.DepositRequest{asset} <- fetchAndArchive depositRequestCid
+          (_, AssetDescription{registrar}) <- fetchByKey @AssetDescription asset.id
 
-      nonconsuming CreditAccount : ContractId AssetDeposit
-        with
-          creditAccountRequestCid : ContractId Custody.CreditAccountRequest
-        do
-          Custody.CreditAccountRequest{accountId; asset} <- fetchAndArchive creditAccountRequestCid
-          depositCid <- exerciseByKey @AssetSettlementRule accountId AssetSettlement_Credit with ctrl = provider; ..
-          deposit <- fetch depositCid
-          exercise depositCid AssetDeposit.AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
+          create AssetDeposit with asset; account; lockers = empty; observers = fromList [ registrar ]
 
-      nonconsuming DebitAccount : Asset
+      nonconsuming Withdrawal : ()
         with
-          debitAccountRequestCid : ContractId Custody.DebitAccountRequest
+          withdrawalRequestCid : ContractId Custody.WithdrawalRequest
         do
-          Custody.DebitAccountRequest{accountId; debit} <- fetchAndArchive debitAccountRequestCid
-          exerciseByKey @AssetSettlementRule accountId debit
-
-      nonconsuming TransferDeposit : ContractId AssetDeposit
-        with
-          transferDepositRequestCid : ContractId Custody.TransferDepositRequest
-        do
-          Custody.TransferDepositRequest{accountId; transfer} <- fetchAndArchive transferDepositRequestCid
-          exerciseByKey @AssetSettlementRule accountId transfer
+          Custody.WithdrawalRequest{depositCid} <- fetchAndArchive withdrawalRequestCid
+          archive depositCid
 
```

Comment:
I think we need to assert here as well that the account provider on the asset deposit is in fact the provider of the custody service we're in. The archive would fail if it's not, but with the assert we can provide a proper error message.

Reply:
I've added the assertion in the latest commit.